app/controllers/chats_controller.rb

```diff
@@ -7,11 +7,10 @@ def create
     @sender = current_user
     @recipient = User.find_by(email: chat_params[:email])
     if Chat.between(@sender.id, @recipient.id).present?
-      @chat = Chat.between(@sender.id, @recipient.id)
+      @chat = Chat.between(@sender.id, @recipient.id).first
     else
       @chat = Chat.create({sender_id: @sender.id, recipient_id: @recipient.id})
     end
```

Comment:
I'm not sure I accept that the only way to test this is with factories, though. And I'm not too happy about allowing logic in the controller that shouldn't be there - it leaves a 'broken window' in the code that will need fixing later. 

https://medium.freecodecamp.org/pragmatic-programmer-broken-windows-6916998eecbe